### PR TITLE
fix some edge cases in new sync log logic

### PR DESCRIPTION
**Pull request by czue**
http://manage.dimagi.com/default.asp?177490

@snopoke found some holes in our tests

### CODE REVIEW

**Comment by czue** [comment]
wait for build

File: corehq/ex-submodules/casexml/apps/case/mock.py [hunk 1]
```diff
@@ -186,9 +187,10 @@ def __init__(self,
 
 
         # fail if user specifies both, say, case_name='Johnny' and update={'case_name': 'Johnny'}
-        for key in create_or_update:
-            if create_or_update[key] is not CaseBlock.undefined and self['update'].has_key(key):
-                raise CaseBlockError("Key %r specified twice" % key)
+        if strict:
+            for key in create_or_update:
+                if create_or_update[key] is not CaseBlock.undefined and self['update'].has_key(key):
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/mock.py]: W601 .has_key() is deprecated, use 'in'

**Comment by snopoke** [comment]
:+1:

Thought: if I create a case with type='edge' does that make it an edge case?

**Comment by czue** [comment]
:sunglasses:

**Comment by benrudolph** [comment]
lol

**Comment by czue** [comment]
:white_check_mark:
